### PR TITLE
Stop inserting semicolon when extracting match arm

### PR DESCRIPTION
**Pull request by omertuc**
# Overview

Extracting a match arm value that has type unit into a function, when a
comma already follows the match arm value, results in an invalid (syntax
error) semicolon added between the newly generated function's generated
call and the comma.

# Example

Running this extraction

```rust
fn main() {
    match () {
        _ => $0()$0,
    };
}
```

would lead to

```rust
fn main() {
    match () {
        _ => fun_name();,
    };
}

fn fun_name() {
}
```

# Issue / Fix details

This happens because when there is no comma, rust-analyzer would simply
add the comma and wouldn't even try to evaluate whether it needs to add
a semicolon. But when the comma is there, it proceeds to evaluate
whether it needs to add a semicolon and it looks like the evaluation
logic erroneously ignores the possibility that we're in a match arm.
IIUC it never makes sense to add a semicolon when we're extracting from
a match arm value, so I've adjusted the logic to always decide against
adding a semicolon when we're in a match arm

### CODE REVIEW

**Review by lowr — COMMENTED** [review]
Nice find! Looks good to me modulo one nit.

File: crates/ide-assists/src/handlers/extract_function.rs [hunk 1]
```diff
@@ -1360,14 +1360,15 @@ fn make_call(ctx: &AssistContext<'_>, fun: &Function, indent: IndentLevel) -> St
     }
 
     format_to!(buf, "{expr}");
-    let insert_comma = fun
-        .body
-        .parent()
-        .and_then(ast::MatchArm::cast)
-        .map_or(false, |it| it.comma_token().is_none());
+    let parent_match_arm = fun.body.parent().and_then(ast::MatchArm::cast);
+    let insert_comma = parent_match_arm.clone().map_or(false, |it| it.comma_token().is_none());
```
Comment (lowr) [inline crates/ide-assists/src/handlers/extract_function.rs:1364]:
```suggestion
    let insert_comma = parent_match_arm.as_ref().is_some_and(|it| it.comma_token().is_none());
```

**Comment by omertuc** [comment]
Force pushed to apply suggestion

**Comment by lowr** [comment]
Thanks!

@bors r+

**Pull request merged**